project_task_reply_stage/models/mail_message.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2022 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class MailMessage(models.Model):
+    _inherit = "mail.message"
+
+    @api.model_create_multi
+    def create(self, values_list):
+        messages = super(MailMessage, self).create(values_list)
+        for message in messages:
+            if not (
+                message.model == "project.task"
+                and message.subtype_id
+                and not message.subtype_id.internal
+            ):
+                continue
+            task = self.env["project.task"].browse(message.res_id)
+            reply_stage = self.env["project.task.type"].browse(
+                int(
+                    self.env["ir.config_parameter"]
+                    .sudo()
+                    .get_param("project_task_reply_stage.customer_reply_stage_id")
+                )
+            )
+            if not (
+                task.state != "draft"
+                and (
+                    not message.author_id.user_ids
+                    or (
+                        message.author_id.user_ids
+                        and not message.author_id.user_ids[0].has_group(
+                            "base.group_user"
+                        )
+                    )
+                )
+            ):
+                continue
+            if task.project_id and task.project_id.customer_reply_stage_id:
+                task.sudo().write(
+                    {"stage_id": task.project_id.customer_reply_stage_id.id}
+                )
+            else:
+                if task.project_id in reply_stage.project_ids:
+                    task.sudo().write({"stage_id": reply_stage.id})
```

Comment:
```suggestion
            if task.state == "draft":
                continue
            user = message.author_id.user_ids[:1]
            if not (user and user.has_group("base.group_user")):
                continue
            project = task.project_id
            if project and project.customer_reply_stage_id:
                task.sudo().write({"stage_id": project.customer_reply_stage_id.id})
                continue
            reply_stage_id = int(
                self.env["ir.config_parameter"]
                .sudo()
                .get_param("project_task_reply_stage.customer_reply_stage_id")
            )
            if reply_stage_id in project.type_ids.ids:
                task.sudo().write({"stage_id": reply_stage_id})
```
Just made the code concise.  Please confirm that this is valid.